include/openssl/e_ostime.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 1995-2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OPENSSL_E_OSTIME_H
+# define OPENSSL_E_OSTIME_H
+# pragma once
```

Comment:
(aside) I know this is standard OpenSSL practice, but why do we put `#pragma once` inside the include guards? Both provide the same protection against multiple inclusion, and there are anecdotal claims that `pragma once` may be faster on some compilers - but in that case, shouldn't it be first? Is it a belt-and-braces check in case we get the include guard macro name on one of the lines?

Reply:
>  standard OpenSSL practice

To be honest, it's not a longstanding practice. Most of the pragmas were added in #9333 using an automated script.

> anecdotal claims that pragma once may be faster on some compilers

The improvement is that the compiler itself remembers not to open the header file over and over again, just to let the preprocessor read and skip the entire content. The perfomance benefit might not be as significant as it used to be, due to sophisticated caching mechanisms, but still it's a simpler and more comprehensible approach than the common preprocessor-based pattern (which is an ancient preprocessor workaround for a non-existing compiler feature imho),  and less intrusive than using [external include guards](https://github.com/openssl/openssl/pull/9333#issuecomment-509934093) to improve performance.

>  - but in that case, shouldn't it be first?

In theory, I agree. But unless somebody uses external include guards, it wouldn't make a practical difference.

The reason why I added them inside the include guards and not at the very beginning is simply because that was how I remembered seeing it in Microsoft headers, at least in MFC (Microsoft Foundation Classes) headers many years ago.

A quick glance at my Visual Studio 2022 headers shows that this seems not to be the case anymore.
```C
//
// stdio.h
//
//      Copyright (c) Microsoft Corporation. All rights reserved.
//
// The C Standard Library <stdio.h> header.
//
#pragma once
#ifndef _INC_STDIO // include guard for 3rd party interop
#define _INC_STDIO
```


See also
- https://github.com/openssl/openssl/pull/9333#issuecomment-511129392
- https://github.com/openssl/openssl/pull/14096#issuecomment-775554386